imagedephi/redact.py

```diff
@@ -1,348 +0,0 @@
-from __future__ import annotations
-
-import abc
-from collections.abc import Generator
-import importlib.resources
-from itertools import chain
-from pathlib import Path
-from typing import TYPE_CHECKING
-
-import click
-import tifftools
-import tifftools.constants
-import yaml
-
-from imagedephi.rules import (
-    FileFormat,
-    MetadataSvsRule,
-    MetadataTiffRule,
-    RuleSet,
-    RuleSource,
-    SvsDescription,
-    build_ruleset,
-)
-
-if TYPE_CHECKING:
-    from tifftools.tifftools import IFD, TiffInfo
-
-
-FILE_EXTENSION_MAP: dict[str, FileFormat] = {
-    ".tif": FileFormat.TIFF,
-    ".tiff": FileFormat.TIFF,
-    ".svs": FileFormat.SVS,
-}
-
-
-class MalformedAperioFileError(Exception):
-    """Raised when the program cannot process an Aperio/SVS file as expected."""
-
-    ...
-
-
-class RedactionPlan:
-    file_format: FileFormat
-
-    @abc.abstractmethod
-    def report_plan(self) -> None:
-        ...
-
-    @abc.abstractmethod
-    def execute_plan(self) -> None:
-        ...
-
-    @abc.abstractmethod
-    def is_comprehensive(self) -> bool:
-        """Return whether the plan redacts all metadata and/or images needed."""
-        ...
-
-    @abc.abstractmethod
-    def report_missing_rules(self) -> None:
-        ...
-
-
-class TiffBasedMetadataRedactionPlan(RedactionPlan):
-    tiff_info: TiffInfo
-
-    @classmethod
-    def build(
-        cls, image_path: Path, base_rules: RuleSet, override_rules: RuleSet | None = None
-    ) -> TiffBasedMetadataRedactionPlan:
-        file_extension = image_path.suffix
-        if FILE_EXTENSION_MAP[file_extension] == FileFormat.TIFF:
-            tiff_info = tifftools.read_tiff(str(image_path))
-            return TiffMetadataRedactionPlan(
-                tiff_info,
-                base_rules.get_metadata_tiff_rules(),
-                override_rules.get_metadata_tiff_rules() if override_rules else [],
-            )
-        if FILE_EXTENSION_MAP[file_extension] == FileFormat.SVS:
-            tiff_info = tifftools.read_tiff(str(image_path))
-            return SvsMetadataRedactionPlan(
-                tiff_info,
-                base_rules.get_metadata_tiff_rules(),
-                base_rules.get_metadata_svs_rules(),
-                override_rules.get_metadata_tiff_rules() if override_rules else [],
-                override_rules.get_metadata_svs_rules() if override_rules else [],
-            )
```

Comment:
Notice how the two redaction plan classes have different signatures. This was changed as part of the reorg.